app/src/main/java/io/appium/settings/receivers/LocaleSettingReceiver.java

```diff
@@ -46,10 +48,21 @@ public void onReceive(Context context, Intent intent) {
         String country = intent.getStringExtra(COUNTRY);
 
         // Expect https://developer.android.com/reference/java/util/Locale.html#Locale(java.lang.String,%20java.lang.String) format.
-        // If we'd like to extend the other format like `Locale(String language, String country, String variant)`,
-        // you can implement it around here.
         Locale locale = new Locale(language, country);
 
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            String script = intent.getStringExtra(SCRIPT);
+
+            Locale.Builder builder = new Locale.Builder();
+            builder.setLocale(locale);
+            builder.setScript(script == null ? "" : script); // "Hans" part
```

Comment:
is it ok to set it to an empty string?

Reply:
Yes. Internally, `Locale` handles the `script` as `""` as default. Other methods also set the part with `""` if they have no `sctipt` value.